util/EIP-1024/src/lib.rs

```diff
@@ -0,0 +1,139 @@
+extern crate saltbabe;
+#[macro_use]
+extern crate arrayref;
+extern crate hex;
+use std::str;
+pub use saltbabe::{KeyPair,Public, Secret, Error};
+pub use saltbabe::traits::FromUnsafeSlice;
+
+const VERSION: &str =  "x25519-xsalsa20-poly1305";
+
+#[derive(Debug, Clone)]
+pub struct EncryptedData {
+    version: String,    
+    nonce: String,
+    ephem_public: String,
+    ciphertext: String
+}
+
+pub fn get_encryption_keypair(sk: [u8; 32]) -> KeyPair<Secret, Public> {
+    let keypair = saltbabe::crypto_box::gen_keypair_from_secret(&sk);
+    return keypair;
+}
+
+pub fn to_byte32(bytes: &[u8]) -> [u8; 32] {
+    array_ref!(bytes, 0, 32).clone()
```

Comment:
This use of `array_ref` is questionable, why not just copy to the bytes into `[u8; 32]`? 

Also, the `clone()` is not needed because `&[u8; 32]` is `Copy` then use `deref` instead

Reply:
According to the docs `array_ref!` will panic if too short slice is provided. I think we should rather fill it with zeros or just return a `Result`.
I think @niklasad1 suggestion to just copy to an array is a good one.

Reply:
Ok, I should add that.